test/models/profile_test.rb

```diff
@@ -235,65 +312,57 @@ class ProfileTest < ActiveSupport::TestCase
-    p1 = Profile.create!(ref_id: 'p1_foo_ref_id',
-                         name: 'p1 foo',
-                         benchmark_id: benchmarks(:one).id,
-                         parent_profile_id: profiles(:one).id)
-    p2 = Profile.create!(ref_id: 'p2_foo_ref_id',
-                         name: 'p2 foo',
-                         benchmark_id: benchmarks(:one).id)
+    p1 = FactoryBot.create(:profile, account: @account)
+    p2 = FactoryBot.create(:canonical_profile)
     assert_includes Profile.canonical, p2
     assert_not_includes Profile.canonical, p1
   end
 
   context 'in_policy scope' do
     setup do
-      profiles(:one).update!(policy: policies(:one),
-                             account: accounts(:one))
+      @profile = FactoryBot.create(:profile, account: @account, policy: @policy)
     end
 
     should 'find by exact profile id' do
-      profiles(:two).update!(account: accounts(:one))
-      assert_includes Profile.in_policy(profiles(:two).id),
-                      profiles(:two)
-      assert_equal 1, Profile.in_policy(profiles(:two).id).length
+      assert_includes Profile.in_policy(@profile.id), @profile
+      assert_equal 1, Profile.in_policy(@profile.id).length
     end
 
     should 'find all policy profiles with policy id provided' do
-      profiles(:two).update!(policy: policies(:one),
-                             external: true,
-                             account: accounts(:one))
-
-      profiles(:two).dup.update!(policy: policies(:two),
-                                 account: accounts(:one))
+      p2 = FactoryBot.create(
+        :profile,
+        parent_profile: @profile.parent_profile,
+        external: true,
+        account: @account,
+        policy: @policy
+      )
 
-      returned_profiles = Profile.in_policy(policies(:one).id)
-      assert_includes returned_profiles, profiles(:one)
-      assert_includes returned_profiles, profiles(:two)
+      returned_profiles = Profile.in_policy(@policy.id)
+      assert_includes returned_profiles, @profile
+      assert_includes returned_profiles, p2
       assert_equal 2, returned_profiles.length
     end
 
     should 'find all policy profiles with any policy profile id provided' do
       # set different UUIDs on profiles, as they share
       # the same labels/uuids with policies.
-      profiles(:one).update!(id: SecureRandom.uuid)
-      profiles(:two).update!(id: SecureRandom.uuid,
-                             policy: policies(:one),
-                             external: true,
-                             account: accounts(:one))
-
-      profiles(:two).dup.update!(policy: policies(:two).dup,
-                                 account: accounts(:one))
-
-      returned_profiles = Profile.in_policy(profiles(:two).id)
-      assert_includes returned_profiles, profiles(:one)
-      assert_includes returned_profiles, profiles(:two)
+      p2 = FactoryBot.create(
+        :profile,
+        id: SecureRandom.uuid,
```

Comment:
Observation: These UUID generations were in place here to overcome how the previous Rails factories has generated them. I guess this would be fixed with the use of FactoryBot, and every new record would have a unique UUID.